Documents/Technical Specifications/Technical.md

```diff
@@ -0,0 +1,494 @@
+
+## Unreal Engine
+
+We must use Unreal Engine 4.27.
+
+## Github
+
+We must use Github to manage our project.
+We also need to expand storage space for our project ( due to the size of the assets ).
+
+- Perforce doesn't solve the problem (Files still to big for Github).
+- Self-hosted Gitlab doesn't solve the problem (Won't appear on Github).
+- Self-hosted Github doesn't solve the problem (Won't appear on Github).
+- Private External SSD doesn't solve the problem (Won't appear on Github and limit to 1 person working).
+
+# Convention
+
+## Naming
+
+Folders, files, Class, functions, variables must be named in English and in Pascal Snake Case.
+Everythings must be in English and have a meaningfull name.
+
+## Folder / File
+
+Imported assets will be in their own folder.
+Created blueprints will be in 'Blueprint' folder.
+Created materials will be in 'Material' folder.
+Created textures will be in 'Texture' folder.
+Created animations will be in 'Animation' folder.
+Created sounds will be in 'Sound' folder.
+
+## Blueprint
+
+Blueprints must be named in English and in Pascal Snake Case.
+They will be in the 'Blueprint' folder. Except for blueprints imported with an asset.
+
+## Git
+
+### Branch
+
+Each branch must be named in English and have a meaningfull name.
+There will be a branch for:
+
+- The main branch
+- MVP Branch
+- Demo branch
+- One for each functionnalities
+
+### Commit
+
+Each commit must be named in English and have a meaningfull name.
+Multiple commits can be done in one branch.
+Commit as much as possible, it will help us to keep track of the changes.
+You can commit not working code, as long as it's not pushed.
+
+### Pull Request
+
+Each pull request must be named in English and have a meaningfull name.
+Each pull request must be reviewed and approved by at least one other person.
+Each pull request must be merged by the person who created the pull request.
+Each pull request must have a description of the changes made.
+
+### Issue
+
+Each issue must be named in English and have a meaningfull name.
+Each issue must be assigned to a person.
+Each issue must have a description of the problem (c.f [Issues](https://github.com/algosup/2022-2023-project-4-game-design-Team-5/issues)).
+
+# Asset
+
+## Environment
+
+The house asset is from the Unreal Engine Marketplace. It is a triplex house with a lot of rooms and a garden. The asset is free to use for non-commercial projects. The asset is available [here](https://www.unrealengine.com/marketplace/en-US/product/big-triplex-house-villa).
+It's already made and we can use it as a base for our game.
+All materials and textures are already made and applied to the asset.
+
+## Character
+
+### Sube
+
+Sube is the main character and is a sugar cube.
+For the MVP, Sube will be a simple cube with a material of sugar applied to it.
+For the final version, Sube will be a 3D model of a sugar cube with a material of sugar applied to it. It will have to be designed by us on Blender.
```

Comment:
```suggestion
For the final version, Sube will be a 3D model of a sugar cube with a material of sugar applied to it. It will have to be designed by our team on Blender.
```